include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,603 @@
+ * results in the corresponding output packets. Outputted packets contain
+ * compression results metadata (odp_comp_packet_result_t), which should be
+ * checked for operation status.
+ *
+ * For every input packet entry in 'pkt_in' array, application should pass
+ * corresponding valid output packet handle. If any error occurs during
+ * processing of packets, the API returns with number of entries successfully
+ * processed. When hashing is configured along with compression operation the
+ * result is appended at the end of the output packet.
+ *
+ * Processed data length can be computed by subtracting 'digest_len' from
+ * output_data_range.len where 'digest_len' can be queried from
+ * odp_comp_hash_alg_capability().
+ *
+ * @param          pkt_in	Packets to be processed
+ * @param          pkt_out	Packet handle array for resulting packets
+ * @param          num_pkt	Number of packets to be processed
+ * @param          param	Operation parameters
+ *
+ * @return Number of input packets consumed (1 ... num_pkt)
+ * @retval <0 on failure
+ *
+ * @note Both 'pkt_in' and 'pkt_out' array should be of same length, Results
+ * are undefined if otherwise.
+ */
+int odp_comp_op(const odp_packet_t pkt_in[], odp_packet_t pkt_out[],
+		int num_pkt, const odp_comp_packet_op_param_t param[]);
+
+/**
+ * Asynchronous packet compression  operation
+ *
+ * This operation does packet compression in asynchronous mode. It processes
+ * packets otherwise identical to odp_comp_op(), but the resulting packets are
+ * enqueued to 'compl_queue' configured during session (odp_comp_session_t)
+ * creation. For every input packet entry in in_pkt array, user should pass
+ * corresponding valid output packet handle. On return, API returns with
+ * number of entries successfully submitted for operation.
+ *
+ * In case of partially accepted array i.e.
+ * when number of packets returned < num_pkt, application may attempt to
+ * resubmit subsequent entries via calling any of the operation API.
+ *
+ * Processed data length can be computed by subtracting 'digest_len' from
+ * output_data_range.len where 'digest_len' can be queried from
+ * odp_comp_hash_alg_capability().
+ *
+ * All the packets successfully enqueue will be submitted to 'compl_queue'
+ * after compression operation, Application should check 'status' of the
+ * operation in odp_comp_packet_result_t.
+ *
+ * Please note it is always recommended that application using async mode,
+ * provide sufficiently large buffer size to avoid
+ * ODP_COMP_ERR_OUT_OF_SPACE_TERM.
+ *
+ * @param	pkt_in		Packets to be processed
+ * @param	pkt_out		Packet handle array for resulting packets
+ * @param	num_pkt		Number of packets to be processed
+ * @param	param		Operation parameters
+ *
+ * @return	Number of input packets enqueued (0 ... num_pkt)
+ * @retval	<0 on failure
+ *
+ * @note Both 'pkt_in' and 'pkt_out' array should be of same length, Results
+ * are undefined if otherwise.
```

Comment:
Again, shouldn't `odp_comp_packet_op_param_t` array also be the same length?

Reply:
Yes. Updated